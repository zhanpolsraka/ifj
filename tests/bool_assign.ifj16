//0
class Main{
	static boolean a = "true";
	static void run(){
		boolean b = Main.a;
	}
}