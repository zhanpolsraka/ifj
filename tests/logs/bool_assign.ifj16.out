---------------FILENAME: tests/bool_assign.ifj16---------------


Instruction stack item [7] -> class with name "Main"
Instruction stack item [6] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [5] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [4] -> function with name "run"
Instruction stack item [3] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [2] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [1] -> end function
Instruction stack item [0] -> end class


---------------MAKE RELATIONS--------------
                     *
                     *
                     *
------------------EXECUTE------------------

------------------OUTPUT-------------------

STEP 1 - non-func instructions
Found Main
STEP 1 - non-func instructions
INSERT 1 (boolean)
---------------------------------------------------
N0: type is 26, values is 1 (bool) 
---------------------------------------------------
STEP 1 - non-func instructions
ASSIGN 1 (boolean) to 0xc9a400
---------------------------------------------------
N0: type is 26, values is 1 (bool) 
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
STEP 1 - non-func instructions
Got in function - ignore instructions until I get out
Found run()
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
Got out of function
Got out of run()
STEP 1 - non-func instructions
Got out of Main
STEP 2 - validating existance of run() in Main
STEP 3 - executing run() instructions
INSERT 1 (boolean)
---------------------------------------------------
N0: type is 26, values is 1 (bool) 
---------------------------------------------------
ASSIGN 1 (boolean) to 0xc9a520
---------------------------------------------------
N0: type is 26, values is 1 (bool) 
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------

-----------------END OUTPUT----------------


----------------ALL RIGHT!!!---------------

----------------SYMBOL TREE----------------

Class Main
	Variable -> a, type BOOLEAN, value [1]
	Function -> run
Lokal variables: 
	Variable -> b, type BOOLEAN, value [1]


--------------INSTRUCTION LIST-------------

Instruction stack item [7] -> class with name "Main"
Instruction stack item [6] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [5] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [4] -> function with name "run"
Instruction stack item [3] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [2] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [1] -> end function
Instruction stack item [0] -> end class



